lib/Models/Catalog/CatalogItems/CartoMapV3CatalogItem.ts

```diff
@@ -0,0 +1,290 @@
+      });
+    }
+
+    let geoJsonUrls: string[] = [];
+    let mvtTileJsonUrls: string[] = [];
+
+    if (
+      response &&
+      isJsonObject(response.geojson) &&
+      isJsonStringArray(response.geojson.url)
+    ) {
+      geoJsonUrls = response.geojson.url;
+    }
+
+    if (
+      response &&
+      isJsonObject(response.tilejson) &&
+      isJsonStringArray(response.tilejson.url)
+    ) {
+      mvtTileJsonUrls = response.tilejson.url;
+    }
+
+    if (geoJsonUrls.length === 0 /*&& mvtTileJsonUrls.length === 0*/) {
+      throw TerriaError.from("No GeoJSON found.");
+    }
+
+    runInAction(() => {
+      if (response && isJsonNumber(response?.size)) {
+        this.geoJsonSize = response.size;
+      }
+      this.geoJsonUrls = geoJsonUrls;
+      this.mvtTileJsonUrls = mvtTileJsonUrls;
+    });
+  }
+
+  protected async forceLoadGeojsonData() {
+    if (this.geoJsonUrls.length === 0)
+      throw TerriaError.from("No GeoJSON URL found for Carto table");
+
+    let jsonData: JsonValue | undefined = undefined;
+
+    // Download all geoJson files
+    const geojsonResponses = await Promise.all(
+      this.geoJsonUrls.map(async (url) => {
+        jsonData = (await callCartoApi(url, this.accessToken)).throwIfError();
+
+        if (jsonData === undefined) {
+          throw new TerriaError({
+            title: "Failed to load GeoJSON",
+            message: `Failed to load GeoJSON URL ${url}`
+          });
+        }
+
+        if (
+          isJsonObject(jsonData, false) &&
+          typeof jsonData.type === "string"
+        ) {
+          // Actual geojson
+          const fc = toFeatureCollection(jsonData);
+          if (fc) return fc;
+        }
+
+        throw new TerriaError({
+          title: "Failed to load GeoJSON",
+          message: `Invalid response from GeoJSON URL ${url}:\n\n
+          \`\`\`
+          ${JSON.stringify(jsonData)}
+          \`\`\``
+        });
+      })
+    );
+
+    // NOTE: Commented out until we add tileJson/mvt support
+    // Download all tileJson files
+    // const tilejsonResponses = await Promise.all(
+    //   this.mvtTileJsonUrls.map(async (url) => {
+    //     jsonData = await loadJson(url, {
+    //       Authorization: `Bearer ${this.accessToken}`
+    //     });
+
+    //     if (jsonData === undefined) {
+    //       throw new TerriaError({
+    //         title: "Failed to load GeoJSON",
+    //         message: `Failed to load GeoJSON URL ${url}`
+    //       });
+    //     }
+
+    //     if (isJsonObject(jsonData, false)) {
+    //       return jsonData;
+    //     }
+
+    //     throw new TerriaError({
+    //       title: "Failed to load GeoJSON",
+    //       message: `Invalid response from GeoJSON URL ${url}:\n\n
+    //       \`\`\`
+    //       ${JSON.stringify(jsonData)}
+    //       \`\`\``
+    //     });
+    //   })
+    // );
+
+    // Merge all geojson responses into a combined feature collection
```

Comment:
This should always be fine hey? Any situation where we would not want to combine all the geojson into one Feature collection? Would sometimes a request to Carto return different version of the same data, and therefore combining it would lead to duplicates?

Reply:
I actually couldn't come across any examples of the Carto API returning geoJSON URLs - but it does return an array of URLs.

I figured the best default behaviour would be to merge them. Alternatively, I can just use the first geoJSON URL - or maybe throw an error if multiple URLs are returned

Reply:
I think merge is probably best actually. Cheers